core/include/atomic.h

```diff
@@ -35,9 +46,101 @@
  */
 unsigned int atomic_set_return(unsigned int *val, unsigned int set);
```

Comment:
I just checked, looks like there are only two or three uses of ```atomic_set_return```. We should (maybe as a follow-up PR) rewrite this to use ```atomic_int_t``` and update the places where it is used.